### DISCUSSION TITLE
Issue with Overwriting Default Exclude Patterns in Coverage Configuration

### DISCUSSION THREAD

**Discussion by Evgenios95** [vitest-dev/vitest#5462] [Q&A] [answered]
Hey there! I've been working on integrating Vitest into our project for test coverage, and I bumped into a bit of a hiccup with the coverage configuration, particularly around excluding files and folders.

Our project has a specific need to exclude certain paths from the coveragestirues. To achieve this, I added these patterns to the exclude array in our Vitest configuration, like so:

```
export default {
  // Other configuration...
  test: {
    // Attempting to exclude storybook files (not sure if the path is correct now, but lets assume it is)
    exclude: ['**/*.stories.ts'],
    coverage: {
      reporter: ['html'],
      enabled: true
    }
  }
}
```

However, after doing this, I noticed that the defaults usually excluded by Vitest (as mentioned in the [official documentation](https://vitest.dev/config/#coverage-exclude)) are no longer automatically excluded. It seems like specifying any custom paths in the exclude array overrides these defaults completely.

This isn't ideal since I'd prefer not to manually list all the default patterns again in my configuration just to keep them excluded. I was hoping there might be a way to add to the existing defaults rather than replacing them entirely.

Is this a known issue, or is there perhaps a workaround I'm not aware of? It would be great to ensure the defaults are maintained while still being able to specify additional paths to exclude. Thanks for any guidance you can provide!

**Comment by AriPerkkio** [comment, marked as answer]
You'll need to extend the default values. I was just updating this to the documentation: https://github.com/vitest-dev/vitest/pull/5457/commits/59c71d2c1130b69f6a54bd6beab8add6183864a2

> This option overrides all default options. Extend the default options when adding new patterns to ignore:
> 
> ```ts
> import { coverageConfigDefaults, defineConfig } from 'vitest/config'
> 
> export default defineConfig({
>   test: {
>     coverage: {
>       exclude: ['**/custom-pattern/**', ...coverageConfigDefaults.exclude]
>     },
>   },
> })
> ```
>

**Reply by Evgenios95** [reply]
That's awesome! Thanks for the info :)
